mrmustard/lab_dev/circuit_components.py

```diff
@@ -454,7 +453,7 @@ def to_fock(self, shape=None):
         fock._original_bargmann_data = self.representation.data
         return self._from_attributes(fock, self.wires, self.name)
 
-    def auto_shape(self, **kwargs) -> tuple[int, ...]:
+    def auto_shape(self, **_) -> tuple[int, ...]:
```

Comment:
Hmm shouldnt the kwargs be removed if its not used?

Reply:
perhaps.. tbh I'm not sure why codefactor didn't catch this in Filippo's PR so I just made a judgement call here. I decided to do this because DM and Ket (child classes of this one) re-implement it, and I don't think we need a type error if somehow the super() implementation gets called with the kwargs propagated.

Reply:
I added them because auto_shape exists in Ket, DM and CircuitComponent, but in CircuitComponent it cannot use the arguments that can be passed to the version in Ket and DM, but I didn't want it to hang in case the args are kept and it's called on non-states. I won't die on this hill 😅

Reply:
Looks like the proper thing to do is to just keep the **kwargs https://stackoverflow.com/questions/6034662/python-method-overriding-does-signature-matter

Reply:
will leave it as-is then 👍 (I changed it from `kwargs` to `_` because pylint complains that you're not using it, and underscore is the conventional placeholder var name)